incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -105,6 +106,52 @@ var _ = Describe("HNCConfiguration", func() {
 		// is not added to HNCConfiguration.
 		Expect(hasResourceQuota(ctx, barName, "foo-resource-quota")).Should(BeFalse())
 	})
+
+	It("should not propagate objects if the mode of a type is ignore", func() {
+		setParent(ctx, barName, fooName)
+		makeSecret(ctx, fooName, "foo-sec")
+		makeResourceQuota(ctx, fooName, "foo-resource-quota")
```

Comment:
What is the mode here when the secret and resourceQuota are created in parent?

Reply:
Secret is "propagate" because it is set as that in previous test case. ResourceQuota hasn't been configured yet.

Reply:
Discussed offline. Moved `AddHNCConfig` at the beginning to make it clear what the mode will be for each type in the test case.

I will have a followup PR to change all mode of object reconcilers to 'ignore' if the corresponding type is no longer in the `config` singleton.